### ISSUE TITLE
included attributes are not converted to string and `pathing` is not available on `:result`

### ISSUE THREAD

**Issue by leggebroten** [marcdel/open_telemetry_decorator#102]
## Issues
1) OTLP attribute values can only accept strings, integers, doubles, or boolean.
https://hexdocs.pm/opentelemetry_api/OpenTelemetry.html#t:attributes_map/0

Currently if the target of the `:include` is not one of these primitives, the attribute will not added to the span.

2) `:include` cannot handle string-key maps.

3) the `:result` lacks functionality as return values are rarely primitives. 

## Suggested solutions
I would like to submit a PR to address these.
1) do an `inspect` if attribute value is not one of the allowed primitives. 
2) Handle indexing into string-keyed map
3) Add indexing/pathing functionality to `:result`.  e.g. `include: [[:result, :key]]`

**Comment by marcdel** [comment]
Hey, @leggebroten great suggestions! I've been leaning pretty heavily on the fact that Honeycomb somehow magically handles (some) complex objects, but I'm running into it now trying to export them with `Logger.log`.

e.g. enumerables and maps
<img src="https://user-images.githubusercontent.com/693730/235519306-6a3957fa-bfba-44ff-be0d-7327d335c1b8.png" width="35%" height="35%">
<img src="https://user-images.githubusercontent.com/693730/235519580-eb7c6077-e86b-4ad2-ab51-006deb5367db.png" width="35%" height="35%">

Doing an `inspect` on unsupported types seems like a great first step/fallback plan, I'd happily accept a PR for that!

Good call on 2 and 3, I didn't actually realize those didn't work. 3 lines up with a refactor I've been meaning to do, so I went ahead and did that [here](https://github.com/marcdel/open_telemetry_decorator/commit/56c8af0505c47b1e06094541a1e2f0458de8c675). 2 is a little trickier, but if you can find a clean way to do it, I'm down.

I've actually been thinking about making the decorator a little "smarter" with how it handles attributes of certain complex types. For example, lists of primitive types could be joined into a string, maps could be flattened into `thing.other.id=12` as seen above. Then anything we can't figure out gets `inspect`ed so there's at least something in that field. This also might lessen the need to index into string keyed maps.

**Comment by leggebroten** [comment]
Hi @marcdel 
My PR is ready that addresses all three.  I'll try to get it pushed tonight after work.
I like your change to put `:result`.   It simplifies things for me a tiny bit so I'll include it.
I've condensed down the Attributes algorithm a bit and lays groundwork for simply adding arbitrary type conversions ... especially important for `:result`

```
          Kernel.binding()
          |> Keyword.put(:result, result)
          |> Attributes.get(unquote(include))
```

**Comment by marcdel** [comment]
Thanks again, @leggebroten! Didn't realize you already had changes in flight, appreciate you dealing with that rebase.

**Comment by marcdel** [comment]
Closed by #107
